contracts/eosio.msig/src/eosio.msig.cpp

```diff
@@ -91,19 +104,12 @@ void multisig::approve( name proposer, name proposal_name, permission_level leve
 
    transaction_header trx_header = get_trx_header(prop.packed_transaction.data(), prop.packed_transaction.size());
 
-   if ( prop.earliest_exec_time.has_value() ) {
-      if ( prop.earliest_exec_time.value().has_value() ) {
-         return;
-      } else {
-         auto table_op = [](auto&&, auto&&){};
-         if ( approvals_satisfy_trx_authorization(proposer, proposal_name, prop.packed_transaction, table_op) ) {
-            auto prop_it = proptable.find( proposal_name.value );
-            proptable.modify( prop_it, proposer, [&]( auto& p ) {
+   if ( !prop.earliest_exec_time->has_value() ) {
```

Comment:
There was a mistake in this change. What I meant to say was change to something like the following:

```cpp
if( prop.earliest_exec_time.has_value() ) { 
   if( !prop.earliest_exec_time->has_value() ) {
      auto table_op = [](auto&&, auto&&){};
      if( trx_is_authorized(get_approvals_and_adjust_table(get_self(), proposer, proposal_name, table_op), prop.packed_transaction) ) {
         ...
      }
   }
} else {
   check( trx_header.delay_sec.value == 0, "old proposals are not allowed to have non-zero `delay_sec`; cancel and retry" );
}     
```

Also, the fact that the tests pass despite this change means we would benefit from new test case:
   1. Propose a transaction with a `reqauth` action requiring authorization from `alice@active` and a `delay_sec` of 1.
   2. Approve with `alice@active`. Verify that the `earliest_exec_time` is now properly set to one second after the block time including this `approve` action.
   3. Approve with `bob@active`. Verify that the `earliest_exec_time` has not changed.
   4. Verify that you cannot execute prior to the `**earliest_exec_time`.
   5. After the proper time has been reached, verify that you can successfully execute the proposal.

What I believe you should find after making that unit test is that the current code (as of commit 93b7f061b0776d6258ce857375f52219968fb10c) should incorrectly fail at step 3 with the assert error message "old proposals are not allowed to have non-zero ...". After fixing the code as described above, I expect this new unit test to pass.

Reply:
https://github.com/EOSIO/eosio.contracts/blob/ee3cdb17f330aed66e54ef51aa839e4af7c104c6/contracts/eosio.msig/src/eosio.msig.cpp#L129-L140

**Test case:**
https://github.com/EOSIO/eosio.contracts/blob/ee3cdb17f330aed66e54ef51aa839e4af7c104c6/tests/eosio.msig_tests.cpp#L1248-L1344